HPXMLtoOpenStudio/resources/EPvalidator.rb

```diff
@@ -749,28 +767,119 @@ def self.run_validator(hpxml_doc)
         'Quantity' => zero_or_one, # Uses Reference Home if not provided
       },
 
+      # [Pool]
+      '/HPXML/Building/BuildingDetails/Pools/Pool' => {
+        'SystemIdentifier' => one, # Required by HPXML schema
+        'PoolPumps/PoolPump/SystemIdentifier' => one,
+        'PoolPumps/PoolPump/Load[Units="kWh/year"]/Value' => zero_or_one,
+        'Heater[Type="gas fired" or Type="electric resistance"]' => one,
```

Comment:
I was under the impression that that xpath line meant "if it has a heater, it must be gas or electric".

Reply:
That is correct, it does. As written, if you provided a Heater with type "heat pump", it would complain. But if you were to simply change it to optional (`zero_or_one`), it would no longer complain because it would simply find zero instances of a Heater with gas/elec. 

So you need to continue to use `one` but update the xpath to specify that either:
Heater is not provided
or:
Heater is provided and it's one of the specified types

Does that make sense?

Reply:
I see. Yes, it does.